### DISCUSSION TITLE
Dynamic column select with QueryRawUnsafe

### DISCUSSION THREAD

**Discussion by bdcorps** [prisma/prisma#18066] [Q&A]
I want to run a Postgres query that returns columns based on dynamic select criteria. 

Writing an UNSAFE query like this:

```
const select = `"Page".title`
const page: any[] = await prisma.$queryRawUnsafe(`SELECT $1 from "Page" LIMIT 1`, select)
```

returns `[{"?column?":"\"Page\".title"}]`

but this...(which should be the same)

```
const page: any[] = await prisma.$queryRawUnsafe(`SELECT "Page".title from "Page" LIMIT 1`, select)
```

works and returns a valid row.

Any idea how to fix this?

**Comment by ludralph** [comment]
Hi @bdcorps 👋 

In your first query, you're using a parameterized query with a placeholder ($1) to inject the value of select as the select criteria. However, the select variable is already included in the const select = "Page".title"` line, so the placeholder is not necessary.

Here's how you can modify your first query to achieve the desired result:

```js
const select = `"Page".title`
const page: any[] = await prisma.$queryRawUnsafe(`SELECT ${select} from "Page" LIMIT 1`)
````
You can achieve dynamic column selection based on the select value by directly interpolating the select variable into the query string.

Note that queryRawUnsafe can be potentially unsafe if user input is directly used in the query, as it can lead to SQL injection attacks. It's recommended to use parameterized queries whenever possible to mitigate this risk.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
